syntax/repr.rs

```diff
@@ -0,0 +1,45 @@
+use crate::syntax::Atom::{self, *};
+use syn::parse::{Error, Parse, ParseStream, Result};
+use syn::{Ident, LitInt};
+
+#[derive(Copy, Clone, PartialEq)]
+pub enum Repr {
+    Align(u32),
+    Atom(Atom),
+}
+
+impl Parse for Repr {
+    fn parse(input: ParseStream) -> Result<Self> {
+        let begin = input.cursor();
+        let ident: Ident = input.parse()?;
+        if let Some(atom) = Atom::from(&ident) {
+            match atom {
+                U8 | U16 | U32 | U64 | Usize | I8 | I16 | I32 | I64 | Isize if input.is_empty() => {
+                    return Ok(Repr::Atom(atom));
+                }
+                _ => {}
+            }
+        } else if ident == "align" {
```

Comment:
It would be helpful to give a better error message if a repr attribute is written with a non-literal expression, since C++ allows this but Rust does not. `#[repr(align(1 << 6))]`